### PR TITLE
bug-fix(jinja-template): Adding checks for chaos-uid in all jinja templates

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: Udit Gaurav <uditgaurav@gmail.com>

Issue it resolve: https://github.com/litmuschaos/litmus/issues/1292

**What this PR does / why we need it**:
**_This PR is for:_**

- Adding checks for chaos-uid in all jinja templates

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Checklist**

* [x] Does this PR have a corresponding GitHub issue?
* [ ] Have you included relevant README for the chaoslib/experiment with details?
* [ ] Have you added debug messages where necessary? 
* [ ] Have you added task comments where necessary? 
* [ ] Have you tested the changes for possible failure conditions?
* [ ] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [ ] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [ ] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [ ] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [ ] Has the litmusbooks been linted? 

**Special notes for your reviewer**:

### CODE REVIEW

File: chaoslib/litmus/container_kill/containerd_chaos/containerd.j2 [hunk 1]
```diff
@@ -3,13 +3,18 @@ kind: Job
 metadata:
   name: containerd-chaos
   labels:
+    app: crictl
+{% if c_engine is defined and c_engine != '' %}
```
Comment (ispeakc0de) [inline chaoslib/litmus/container_kill/containerd_chaos/containerd.j2]: {% if chaos_uid is defined and chaos_uid != '' %}

File: chaoslib/litmus/cpu_hog/app_cpu_stress.j2 [hunk 2]
```diff
@@ -1,15 +1,21 @@
 apiVersion: batch/v1
 kind: Job
 metadata:
+{% if c_engine is defined and c_engine != '' %}
```
Comment (ispeakc0de) [inline chaoslib/litmus/cpu_hog/app_cpu_stress.j2]: {% if run_id  is defined and run_id  != ' ' %}

File: chaoslib/litmus/platform/gke/node-cpu-hog-job.j2 [hunk 3]
```diff
@@ -1,15 +1,24 @@
 apiVersion: batch/v1
 kind: Job
 metadata:
+{% if c_engine is defined and c_engine != '' %}
   name: node-cpu-hog-{{ chaos_uid }}
```
Comment (ispeakc0de) [inline chaoslib/litmus/platform/gke/node-cpu-hog-job.j2]: Can you please chaos_uid to run_id? and generate run_id before calling this jinja template. Please refer `chaoslib/pumba/network_chaos/pumba_netem_job.j2`

File: chaoslib/pumba/network_chaos/pumba_netem_job.j2 [hunk 4]
```diff
@@ -2,15 +2,19 @@ apiVersion: batch/v1
 kind: Job
 metadata:
   name: pumba-netem-{{ run_id }}
-  labels: 
+  labels:
+{% if c_engine is defined and c_engine != '' %}
```
Comment (ispeakc0de) [inline chaoslib/pumba/network_chaos/pumba_netem_job.j2]: If only one label is present then put label inside if condition like:
{% if c_engine is defined and c_engine != '' %}  
labels:
chaosUID: {{ chaos_uid }}
{% endif %}

If multiple labels are present then you can use :
labels:
{% if c_engine is defined and c_engine != '' %}  
chaosUID: {{ chaos_uid }}
{% endif %}

File: chaoslib/pumba/pumba.j2 [hunk 5]
```diff
@@ -1,15 +1,23 @@
 apiVersion: batch/v1
 kind: Job
 metadata:
+{% if c_engine is defined and c_engine != '' %}
   name: pumba-sig-kill-{{ chaos_uid }}
```
Comment (ispeakc0de) [inline chaoslib/pumba/pumba.j2]: same as above

File: chaoslib/pumba/network_chaos/pumba_netem_job.j2 (hunk 4 above)
Comment (uditgaurav) [reply chaoslib/pumba/network_chaos/pumba_netem_job.j2]: other one is mandatory doesn't depend on anyone so keeping it away from if - else

File: chaoslib/pumba/pumba.j2 [hunk 6]
```diff
@@ -1,15 +1,23 @@
 apiVersion: batch/v1
 kind: Job
 metadata:
-  name: pumba-sig-kill-{{ chaos_uid }}
+{% if run_id is defined and run_id != '' %}
```
Comment (ksatchit) [inline chaoslib/pumba/pumba.j2]: we seem to be getting run_id always

File: chaoslib/powerfulseal/powerfulseal.j2 [hunk 7]
```diff
@@ -28,18 +28,25 @@ kind: Deployment
...
+{% if chaos_uid is defined and chaos_uid != '' %}
     chaosUID: {{ chaos_uid }}
+{% endif %}
 spec:
   selector:
     matchLabels:
       name: powerfulseal
-      chaosUID: {{ chaos_uid }}
+{% if chaos_uid is defined and chaos_uid != '' %}
+    chaosUID: {{ chaos_uid }}
```
Comment (ispeakc0de) [inline chaoslib/powerfulseal/powerfulseal.j2]: indentation

File: chaoslib/powerfulseal/powerfulseal.j2 [hunk 8]
```diff
@@ -28,18 +28,25 @@ kind: Deployment
...
+{% endif %}
   replicas: 1
   template:
     metadata:
       labels:
         name: powerfulseal
-        chaosUID: {{ chaos_uid }}
+{% if chaos_uid is defined and chaos_uid != '' %}
+    chaosUID: {{ chaos_uid }}
```
Comment (ispeakc0de) [inline chaoslib/powerfulseal/powerfulseal.j2]: same as above

File: chaoslib/pumba/pod_failure_by_sigkill.yml [hunk 9]
```diff
@@ -25,6 +25,15 @@
     - set_fact: 
         app_node: "{{ app_node.stdout }}"
 
+      - block:
```
Comment (ispeakc0de) [inline chaoslib/pumba/pod_failure_by_sigkill.yml]: Remove this block

File: chaoslib/litmus/container_kill/containerd_chaos/containerd.j2 [hunk 10]
```diff
@@ -1,15 +1,20 @@
 apiVersion: batch/v1
 kind: Job
 metadata:
-  name: containerd-chaos
+  app: crictl
```
Comment (ispeakc0de) [inline chaoslib/litmus/container_kill/containerd_chaos/containerd.j2]: name field?

File: chaoslib/litmus/disk_fill/disk_fill_job.j2 [hunk 11]
```diff
@@ -3,13 +3,18 @@ kind: Job
 metadata:
   name: disk-fill
```
Comment (ispeakc0de) [inline chaoslib/litmus/disk_fill/disk_fill_job.j2]: run_id suffix?

File: chaoslib/powerfulseal/pod_failure_by_powerfulseal.yml [hunk 12]
```diff
@@ -12,6 +12,15 @@
         engine_ns: "{{ a_ns }}"
         message: "Injecting {{ c_experiment }} chaos on application pod"
       when: "c_engine is defined and c_engine != ''"
+
+    - block:
```
Comment (ispeakc0de) [inline chaoslib/powerfulseal/pod_failure_by_powerfulseal.yml]: pls use job-name=<job-name>-{{run_id}} as a label. refer any other exp

File: chaoslib/powerfulseal/powerfulseal.j2 [hunk 13]
```diff
@@ -28,18 +28,25 @@ kind: Deployment
 metadata:
   name: powerfulseal
   labels:
+    name: powerfulseal
```
Comment (ispeakc0de) [inline chaoslib/powerfulseal/powerfulseal.j2]: run_id prefix?

File: chaoslib/pumba/pod_failure_by_sigkill.yml (hunk 9 above)
Comment (ispeakc0de) [reply chaoslib/pumba/pod_failure_by_sigkill.yml]: It is duplicate

File: chaoslib/pumba/pod_failure_by_sigkill.yml [hunk 14]
```diff
@@ -87,7 +96,7 @@
       
     - name: Wait until the pumba sig-kill job is completed
       shell: >
-        kubectl get pods -l job-name=pumba-sig-kill-{{ chaos_uid }} --no-headers -n {{ namespace }}
+        kubectl get pods -l job-name=pumba-sig-kill-{{ run_id }} --no-headers -n {{ namespace }}
```
Comment (ispeakc0de) [inline chaoslib/pumba/pod_failure_by_sigkill.yml]: pls change the all occurance of chaos_uid to run_id in this file

**Pull request merged**